module/extension/Director/src/Director.hpp

```diff
@@ -373,20 +375,49 @@ namespace module::extension {
          */
         void run_task_pack(const TaskPack& pack);
 
+        /**
+         * A RunReasonLock instance will reset the current_run_reason back to OTHER_TRIGGER when it is destroyed.
+         *
+         * This is used so that the run reason can temporarily be set to something else, and then reset back to the
+         * default if there is an exception or other error.
+         */
+        using RunReasonLock = std::unique_ptr<void, std::function<void(void*)>>;
+
+        /**
+         * Holds a run reason as the current run reason for the current thread.
+         *
+         * It returns a lock object that acts as an RAII object that will reset the current run reason when it is
+         * destroyed.
+         *
+         * @param reason the reason to hold until the lock is destroyed
+         *
+         * @return a lock object that will reset the current run reason when it is destroyed
```

Comment:
```suggestion
         * @return a lock object that will reset the default run reason when it is destroyed
```
This is what it actually does.

Reply:
I meant `current run reason` in terms of the variable `current_run_reason` rather than what I was setting it to, but I can see that there is confusion in that

Reply:
```suggestion
         * @return a lock object that will reset the default current_run_reason when it is destroyed
```